pkg/resource/deploy/plan_executor.go

```diff
@@ -189,13 +191,22 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 	pe.stepExec.WaitForCompletion()
 	logging.V(4).Infof("planExecutor.Execute(...): step executor has completed")
 
+	if res != nil && res.Error() == nil {
+		return res
+	}
```

Comment:
another bail check.

Reply:
i think we should have an idiomatic way of doing this as this form seems a little too low level to me.